numba/_lapack.c

```diff
@@ -546,22 +666,56 @@ numba_xxgetri(char kind, Py_ssize_t n, void *a, Py_ssize_t lda,
             raw_func = get_clapack_zgetri();
             break;
     }
-    if (raw_func == NULL)
-        return -1;
+    ENSURE_VALID_FUNC(raw_func)
 
-    _n = (F_INT) n;
-    _lda = (F_INT) lda;
-    _lwork = (F_INT) lwork[0]; // why is this a ptr?
+    (*(xxgetri_t) raw_func)(&n, a, &lda, ipiv, work, lwork, info);
 
-    (*(xxgetri_t) raw_func)(&_n, a, &_lda, ipiv, work, &_lwork, &_info);
-    *info = (Py_ssize_t) _info;
     return 0;
 }
 
-/* Compute the Cholesky factorization of a matrix.
- * Return -1 on internal error, 0 on success, > 0 on failure.
+/* Compute the inverse of a matrix from the factorization provided by
+ * xxgetrf. (see numba_xxgetrf() about ipiv)
+ * Args are as per LAPACK.
  */
 NUMBA_EXPORT_FUNC(int)
+numba_ez_xxgetri(char kind, Py_ssize_t n, void *a, Py_ssize_t lda,
+                 F_INT *ipiv)
+{
+    F_INT _n, _lda;
+    F_INT lwork = -1;
+    F_INT info = 0;
+    size_t base_size = -1;
+    void * work = NULL;
+    all_dtypes stack_slot;
+
+    ENSURE_VALID_KIND(kind)
+
+    _n = (F_INT)n;
+    _lda = (F_INT)lda;
+
+    base_size = kind_size(kind);
+
+    work = &stack_slot;
+
+    numba_raw_xxgetri(kind, _n, a, _lda, ipiv, work, &lwork, &info);
+    CATCH_LAPACK_INVALID_ARG("xxgetrf", info);
+
+    lwork = cast_from_X(kind, work);
+
+    if (checked_PyMem_RawMalloc(&work, base_size * lwork))
+    {
+        return STATUS_ERROR;
+    }
+
+    numba_raw_xxgetri(kind, _n, a, _lda, ipiv, work, &lwork, &info);
+    CATCH_LAPACK_INVALID_ARG("xxgetrf", info);
```

Comment:
Is this not catching invalid arguments? So simply putting in something invalid would trigger it. So in the above example we could put in `m=7` and `lda=5`.